src/substrait/textplan/SymbolTablePrinter.cpp

```diff
@@ -671,50 +806,293 @@ void SymbolTablePrinter::addInputsToRelation(
+      for (auto& ifThen : *expression->mutable_if_then()->mutable_ifs()) {
+        addInputsToExpression(
+            symbolTable, symbolInfos, ifThen.mutable_if_(), consumedPipelines);
+        addInputsToExpression(
+            symbolTable, symbolInfos, ifThen.mutable_then(), consumedPipelines);
+      }
+      if (expression->if_then().has_else_()) {
+        addInputsToExpression(
+            symbolTable,
+            symbolInfos,
+            expression->mutable_if_then()->mutable_else_(),
+            consumedPipelines);
+      }
+      break;
+    case ::substrait::proto::Expression::kSwitchExpression:
+      if (expression->switch_expression().has_match()) {
+        addInputsToExpression(
+            symbolTable,
+            symbolInfos,
+            expression->mutable_switch_expression()->mutable_match(),
+            consumedPipelines);
+      }
+      for (auto& ifValue :
+           *expression->mutable_switch_expression()->mutable_ifs()) {
+        addInputsToExpression(
+            symbolTable,
+            symbolInfos,
+            ifValue.mutable_then(),
+            consumedPipelines);
+      }
+      if (expression->switch_expression().has_else_()) {
+        addInputsToExpression(
+            symbolTable,
+            symbolInfos,
+            expression->mutable_switch_expression()->mutable_else_(),
+            consumedPipelines);
+      }
+      break;
+    case ::substrait::proto::Expression::kSingularOrList:
+      if (expression->singular_or_list().has_value()) {
+        addInputsToExpression(
+            symbolTable,
+            symbolInfos,
+            expression->mutable_singular_or_list()->mutable_value(),
+            consumedPipelines);
+      }
+      for (auto& option :
+           *expression->mutable_singular_or_list()->mutable_options()) {
+        addInputsToExpression(
+            symbolTable, symbolInfos, &option, consumedPipelines);
+      }
+      break;
+    case ::substrait::proto::Expression::kMultiOrList:
+      for (auto& value :
+           *expression->mutable_multi_or_list()->mutable_value()) {
+        addInputsToExpression(
+            symbolTable, symbolInfos, &value, consumedPipelines);
+      }
+      for (auto& option :
+           *expression->mutable_multi_or_list()->mutable_options()) {
+        for (auto& field : *option.mutable_fields()) {
+          addInputsToExpression(
+              symbolTable, symbolInfos, &field, consumedPipelines);
+        }
+      }
+      break;
+    case ::substrait::proto::Expression::kCast:
+      if (expression->cast().has_input()) {
+        addInputsToExpression(
+            symbolTable,
+            symbolInfos,
+            expression->mutable_cast()->mutable_input(),
+            consumedPipelines);
+      }
+      break;
+    case ::substrait::proto::Expression::kSubquery:
+      // Handled below.
+      break;
+    case ::substrait::proto::Expression::kNested:
+      // TODO -- Implement nested expressions.
```

Comment:
Does this silently fail, e.g. should we have a SUBSTRAIT_FAIL here?

Reply:
Verified that this error is caught in PlanPrinterVisitor.